components/Settings.js

```diff
@@ -340,8 +462,76 @@ class Settings extends React.PureComponent {
     }
   }
 
+  handleChange = (key, value) => {
+    this.props.onChange(key, value)
+    this.props.selectPreset(null)
+    this.setState({ previousSettings: null })
+  }
+
+  getSettings = () =>
+    omit(this.props, [
+      'onChange',
+      'resetDefaultSettings',
+      'applyPreset',
+      'removePreset',
+      'format',
+      'getCarbonImage',
+      'selectPreset',
+      'selectedPreset'
+    ])
+
+  applyPreset = (index, preset) => {
+    const previousSettings = this.getSettings()
+
+    this.props.applyPreset(index, preset)
+    this.setState({ previousSettings })
+  }
+
+  undoPreset = () => {
+    this.props.applyPreset(null, this.state.previousSettings)
+    this.setState({ previousSettings: null })
+  }
+
+  removePreset = index => {
+    this.setState(
+      ({ presets }) => ({ presets: presets.filter((_, i) => i !== index) }),
+      () => {
+        const { selectedPreset, selectPreset } = this.props
+        selectPreset(
+          index === selectedPreset ? null : selectedPreset > 0 ? selectedPreset - 1 : selectedPreset
+        )
+        this.savePresets()
+      }
+    )
+  }
+
+  createPreset = async () => {
+    const preset = this.getSettings()
+
+    preset.custom = true
+
+    preset.icon = await this.props.getCarbonImage({ format: 'png', squared: true })
```

Comment:
Can you also try passing an `exportSize` < 1 to see if we can't cut down on localStorage size? 🙂